openquake/mbt/tests/tools/fault_modeler/test_fault_source_modeler.py

```diff
@@ -0,0 +1,79 @@
+import unittest
+
+import openquake.mbt.tools.fault_modeler.fault_source_modeler as fsm
+
+# -----------------------------------------------------------------------------
+
+param_map = {'source_id': 'ogc_fid',
+             'name': 'ns_name',
+             'average_dip': 'ns_average_dip',
+             'average_rake': 'ns_average_rake',
+             'net_slip_rate': 'ns_net_slip_rate',
+             'vert_slip_rate': 'ns_vert_slip_rate',
+             'strike_slip_rate': 'ns_strike_slip_rate',
+             'shortening_rate': 'ns_shortening_rate',
+             'dip_dir': 'ns_dip_dir',
+             'dip_slip_rate': 'ns_dip_slip_rate'}
+
+geojson_file = 'Data/ne_asia_faults_rates.geojson'
+
+# -----------------------------------------------------------------------------
+
+class TestDatabaseIO(unittest.TestCase):
+
+    def test_fault_database(self):
+
+        # Import the database
+        fault_db = fsm.fault_database()
+        fault_db.import_from_geojson(geojson_file,
+                                      param_map=param_map,
+                                      select_list=[1, 2])
+
+        # Adding a key/value to all faults
+        fault_db.add_property('lower_seismogenic_depth', value=25)
+
+        # Adding key/value to given faults
+        fault_db.add_property('M_max', value=7., id=1)
+        fault_db.add_property('M_max', value=7.5, id=2)
+
+        fault_db.remove_property('name')
+
+        # Esport the augmented database
+        fault_db.export_to_geojson('Data/out_db.geojson')
```

Comment:
No comparison with a reference output ?

Reply:
Added comparison to a reference geojson file. Note that the comparison is not done by simply comparing the two files (e.g. using filecmp), but loading directly the geojson and then comparing the single features, one by one. This was necessary because, for some unknown and strange reason, the json module exports the file with a random order of the internl attributes, which is different each time.

Reply:
I also included OS independent path definition, as indicated in the following comment (so far just for this test)